### PR TITLE
kodi: 17.4 -> 17.6 and move to cmake

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Working on kodi was quite painful as it would rebuild ffmpeg every time. ffmpeg is now built as a separate derivation using the kodi makefile to avoid having to rebuild ffmpeg every time kodi is changed.

Additionally, due to the far superior cmake output a number of dependencies were identified that have been added as well.

This PR compiles but the tests fail (at least no more segfaulting). The main issue with troubleshooting the tests is that they take FOREVER to run now - approximately 4 hours as each test takes 30 seconds. My guess is that there is a timeout of some kind due to our sandbox.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Cc:  @domenkozar @titanous @edwtjo @joachifm

### CODE REVIEW

**Comment by peterhoeg** [comment]
Upgrade to 17.5 works fine but test suite still takes about 4 hours to run (and has a few failures).

@domenkozar @titanous @edwtjo

File: pkgs/applications/video/kodi/default.nix [hunk 1]
```diff
@@ -16,133 +17,194 @@
...
+
+# TODO for Kodi 18.0
+# - cmake is no longer in project/cmake
+# - maybe we can remove auto{conf,make} and libtool from inputs
+# - check if dbus support PR has been merged and add dbus as a buildInput
 
 let
-  kodi_version = "17.4";
+  kodiReleaseDate = "20170422";
+  kodiVersion = "17.4";
```
Comment (domenkozar) [inline pkgs/applications/video/kodi/default.nix]: Shouldn't that be 17.5?

Comment (peterhoeg) [reply pkgs/applications/video/kodi/default.nix]: @domenkozar - it's probably easier for everyone involved if I push the right branch...

**Review by edwtjo — APPROVED** [review]

**Comment by peterhoeg** [comment]
Thanks @edwtjo - this PR disables the checks and I want to be sure everybody is cool with that.

**Comment by peterhoeg** [comment]
Supersedes #31787 if everyone is cool with this.

**Comment by peterhoeg** [comment]
It has a number of security patches (specifically with regards to ffmpeg) - we should probably backport this to 17.09. I'll do later during the weekend.

**Pull request merged**